### PR TITLE
[TestSDK] Add serial_test decorator

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Add serial_test decorator to indicate the test should be run sequentially.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py [hunk 1]
```diff
@@ -5,45 +5,54 @@
...
+
+import pytest
 from azure.cli.testsdk import (LiveScenarioTest, ResourceGroupPreparer, StorageAccountPreparer,
                                JMESPathCheck, JMESPathCheckExists, NoneCheck, api_version_constraint)
 from azure.cli.core.profiles import ResourceType
 
 
 @api_version_constraint(ResourceType.MGMT_STORAGE, min_api='2016-12-01')
 class StorageBlobUploadLiveTests(LiveScenarioTest):
+    @pytest.mark.serial()
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py:17]: May I know the motivation for the change?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py:17]: Check your mail box, Important update of "User Manual of Live Test Pipeline"

**Comment by Juliehzl** [comment]
we could consider integrating the decorator in azure-cli-testsdk and don't fix pytest version in azure cli because it will restrict pytest version once installing azure cli.

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py [hunk 2]
```diff
@@ -8,42 +8,50 @@
 from azure.cli.testsdk import (LiveScenarioTest, ResourceGroupPreparer, StorageAccountPreparer,
                                JMESPathCheck, JMESPathCheckExists, NoneCheck, api_version_constraint)
 from azure.cli.core.profiles import ResourceType
+from azure.cli.testsdk.decorators import serial_test
 
 
 @api_version_constraint(ResourceType.MGMT_STORAGE, min_api='2016-12-01')
 class StorageBlobUploadLiveTests(LiveScenarioTest):
+    @serial_test()
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py:16]: Curious about it. Can we write something like this?
```
@StorageAccountPreparer()
@serial_test()
```

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py:16]: `serial_test` is not like a decorator. Maybe I miss something 0.0

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py:16]: It is a function, not a class. So I use lower case.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_blob_live_scenarios.py:16]: > 
> 
> Curious about it. Can we write something like this?
> 
> ```
> @StorageAccountPreparer()
> @serial_test()
> ```

Good question. Let me try.

File: src/azure-cli-testsdk/azure/cli/testsdk/decorators.py [hunk 3]
```diff
@@ -5,9 +5,18 @@
...
 
     return unittest.skipUnless(get_support_api_version_func()(get_dummy_cli(), resource_type, **kwargs),
                                "Test not supported by current profile.")
+
+
+def serial_test():
+    """
+    Mark the test as serial
+    """
+    return pytest.mark.serial()
```
Comment (jiasli) [inline src/azure-cli-testsdk/azure/cli/testsdk/decorators.py:22]: Parentheses `()` are redundant. See https://docs.pytest.org/en/stable/example/markers.html

**Pull request merged**